EMS/core-bundle/src/Core/ContentType/FieldType/FieldTypeService.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EMS\CoreBundle\Core\ContentType\FieldType;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use EMS\CoreBundle\Entity\ContentType;
+use EMS\CoreBundle\Entity\FieldType;
+use EMS\CoreBundle\Repository\FieldTypeRepository;
+
+class FieldTypeService
+{
+    /** @var ?ArrayCollection<int, FieldType> */
+    private ?ArrayCollection $fieldTypes = null;
+
+    public function __construct(
+        private readonly FieldTypeRepository $fieldTypeRepository
+    ) {
+    }
+
+    public function getTree(ContentType $contentType): FieldTypeTreeItem
+    {
+        return $this->buildTreeItem($contentType->getFieldType());
+    }
+
+    private function buildTreeItem(FieldType $fieldType): FieldTypeTreeItem
+    {
+        $fieldTypeTreeItem = new FieldTypeTreeItem($fieldType);
+
+        foreach ($this->getChildren($fieldType) as $child) {
+            $childTreeItem = $this->buildTreeItem($child)->setParent($fieldTypeTreeItem);
+            $fieldTypeTreeItem->addChild($childTreeItem);
+        }
+
+        $fieldTypeTreeItem->orderChildren();
```

Comment:
Hidden in the FieldTypeTreeItem constructor?

Reply:
https://github.com/ems-project/elasticms/pull/788/commits/eba66f0adc4b1687ecfa837532dcfd0feed934c4